mdetsims/sim_utils.py

```diff
@@ -589,6 +607,9 @@ def _get_band_objects(self):
         positions = []
 
         nobj = self._get_nobj()
+        # turn on/off varying shear
+        if self.flag:
+            shear_red = self.sel.choice(2,nobj,p=[1-self.fraction,self.fraction])
```

Comment:
This line is fine, but a simple choice might be testing if `self.rng.uniform() < frac_ex`.

Reply:
I did it, but somehow the rng did not give constant sequence for the same seed. I could not find the bug causing this, so end up using an independent random number.

Reply:
Hmm. We only need the sims to have the same output given the same seed which should work. I think the issue in the loop is branching based on the flag. We should remove the branch and just check always.